client/my-sites/upgrades/checkout-thank-you/failed-purchase.jsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+import { localize } from 'i18n-calypso';
+import PurchaseDetail from 'components/purchase-detail';
+
+const FailedPurchase = ( { failedPurchases, translate } ) => {
+	const classes = {
+		'checkout-thank-you__header': true,
+	};
+
+	const description = (
+		<div>
+			<p>
+				{
+					translate(
+						'Here\'s the list of products we failed to obtain:'
+					)
+				}
+			</p>
+			<ul className="checkout-thank-you__domain-mapping-details-nameservers">
+				{ failedPurchases.map( item => {
+					return <li>{ item.productName }: { item.meta }</li>
+				} ) }
+			</ul>
+			<p>
+				{
+					translate( 'We have filled your account with credits so you can retry the failed purchases. ' +
+						'If the problem persists, please don\'t hesitate to contact support!'
+					)
+				}
+			</p>
+		</div>
+	);
+
+	return (
+		<div>
+			<div className={ classNames( classes ) }>
```

Comment:
Probably a leftover code (or you'll be expanding it), but just in case: `classes` is constant and seems to always just contain `checkout-thank-you__header` - so you can just use that string here without the need to call `classNames` :)